plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -177,26 +238,135 @@ func (s *SpanReader) FindTraces(queryParams *spanstore.TraceQueryParameters) ([]
 	return retMe, nil
 }
 
-// executeQuery returns UniqueTraceIDs
-func (s *SpanReader) executeQuery(query Query, tableMetrics *casMetrics.Table) (dbmodel.UniqueTraceIDs, error) {
+// FindTraces queries for traces.
+func (s *SpanReader) getTraceIDs(traceQuery *spanstore.TraceQueryParameters) ([]dbmodel.TraceID, error) {
+	if traceQuery.DurationMin != 0 {
+		return s.queryByDuration(traceQuery)
+	}
+
+	// NB(rooz): consider refactoring the query... methods where they take in traceQuery instead of a breakdown of its values
+	if traceQuery.OperationName != "" {
+		traceIds, err := s.queryByServiceNameAndOperation(traceQuery.ServiceName, traceQuery.OperationName, traceQuery.StartTimeMin, traceQuery.StartTimeMax, traceQuery.NumTraces)
```

Comment:
why pass all these parameters to the method if we already have traceQuery container?